sdk/core/Azure.Core/src/Shared/OperationInternalBase.cs

```diff
@@ -188,20 +192,34 @@ private async ValueTask<Response> WaitForCompletionResponseAsync(bool async, Tim
                 return lockOrValue.Value;
             }
 
-            var poller = new OperationPoller(_fallbackStrategy);
-            var response = async
-                ? await poller.WaitForCompletionResponseAsync(this, pollingInterval, cancellationToken).ConfigureAwait(false)
-                : poller.WaitForCompletionResponse(this, pollingInterval, cancellationToken);
+            using var scope = CreateScope(_waitForCompletionScopeName);
```

Comment:
It looks like this will open a scope called `<OperationTypeName>.WaitForCompletion` but the name of the public method is `WaitForCompletionResponse`.  Please update this so the scope name matches the name of the public method.